forum/communications/tests.py

```diff
@@ -1,3 +1,199 @@
 from django.test import TestCase
+from channels.testing import ChannelsLiveServerTestCase
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from rest_framework.test import APITestCase
+from django.contrib.auth import get_user_model
+from .models import Room, Message
 
-# Create your tests here.
+User = get_user_model()
+
+
+class RoomModelTest(TestCase):
+    def setUp(self):
+        self.user1 = User.objects.create_user(
+            email="user1@example.com", password="password"
+        )
+        self.user2 = User.objects.create_user(
+            email="user2@example.com", password="password"
+        )
+        self.room = Room.objects.create(name="room_1_2")
+
+    def test_room_name(self):
+        self.assertEqual(self.room.name, "room_1_2")
+
+    def test_join_user(self):
+        self.room.join(self.user1)
+        self.assertIn(self.user1, self.room.online.all())
+
+    def test_leave_user(self):
+        self.room.join(self.user1)
+        self.room.leave(self.user1)
+        self.assertNotIn(self.user1, self.room.online.all())
+
+    def test_get_users_id(self):
+        self.assertEqual(self.room.get_users_id(), {"user_1": 1, "user_2": 2})
+
+    def test_get_users_names(self):
+
+        self.room.name = f"room_{self.user1.user_id}_{self.user2.user_id}"
+        self.room.save()
+        self.room.online.set([self.user1, self.user2])
+
+        expected_names = {
+            "user_1": self.user1.first_name,
+            "user_2": self.user2.first_name,
+        }
+        self.assertEqual(self.room.get_users_names(), expected_names)
```

Comment:
Add tests for edge cases, e.g., calling 'leave()' for a user not in the room.
Test 'get_online_count()' when no users are in the room.

_Examples_
```
def test_leave_nonexistent_user(self):
    """Ensure no error is raised when a user leaves a room they are not in."""
    self.room.leave(self.user1)
    self.assertNotIn(self.user1, self.room.online.all())

def test_get_online_count_empty_room(self):
    """Ensure online count is zero for an empty room."""
    self.assertEqual(self.room.get_online_count(), 0)
```

Reply:
Done